components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/CertificateMgtUtils.java

```diff
@@ -0,0 +1,170 @@
+                if (trustStore.containsAlias(alias)) {
+                    isCertExists = true;
+                } else {
+                    /*
+                    * If alias is not exists, check whether the certificate is expired or not. If expired set the
+                    * expired flag.
+                    * */
+                    X509Certificate x509Certificate = (X509Certificate) certificate;
+                    if (x509Certificate.getNotAfter().getTime() <= System.currentTimeMillis()) {
+                        expired = true;
+                    } else {
+                        //If not expired add the certificate to trust store.
+                        trustStore.setCertificateEntry(alias, certificate);
+                    }
+                }
+            }
+
+            serverCert.close();
+            OutputStream out = new FileOutputStream(trustStoreFile);
+            trustStore.store(out, TRUST_STORE_PASSWORD.toCharArray());
+            out.close();
+            return expired ? ResponseCode.CERTIFICATE_EXPIRED :
+                    isCertExists ? ResponseCode.ALIAS_EXISTS_IN_TRUST_STORE : ResponseCode.SUCCESS;
+        } catch (CertificateException e) {
+            log.error("Error loading certificate.", e);
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (FileNotFoundException e) {
+            log.error("Error reading/ writing to the certificate file.", e);
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Could not find the algorithm to load the certificate.", e);
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (UnsupportedEncodingException e) {
+            log.error("Error retrieving certificate from String", e);
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (KeyStoreException e) {
+            log.error("Error reading certificate contents.", e);
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        } catch (IOException e) {
+            log.error("Error in loading the certificate.", e);
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        }
+    }
+
+    /**
+     * This method will remove certificate from the trust store which matches the given alias.
+     *
+     * @param alias : The alias which the certificate should be deleted.
+     * @return : ResponseCode based on the execution.
+     *
+     * Response Codes
+     * SUCCESS : If the certificate is deleted successfully.
+     * INTERNAL_SERVER_ERROR : If any exception occurred.
+     * CERTIFICATE_NOT_FOUND : If the Alias is not found in the key store.
+     */
+    public ResponseCode removeCertificateFromTrustStore(String alias) {
+        boolean isExists;
+        try {
+            File trustStoreFile = new File(TRUST_STORE);
+            InputStream localTrustStoreStream = new FileInputStream(trustStoreFile);
```

Comment:
Streams need to close in finally block and use java8 style to open files and do the operations